include/vectorpostprocessors/ResponseHistoryMean.h

```diff
@@ -0,0 +1,62 @@
+/*************************************************/
+/*           DO NOT MODIFY THIS HEADER           */
+/*                                               */
+/*                     MASTODON                  */
+/*                                               */
+/*    (c) 2015 Battelle Energy Alliance, LLC     */
+/*            ALL RIGHTS RESERVED                */
+/*                                               */
+/*   Prepared by Battelle Energy Alliance, LLC   */
+/*     With the U. S. Department of Energy       */
+/*                                               */
+/*     See COPYRIGHT for full restrictions       */
+/*************************************************/
+
+#ifndef RESPONSEHISTORYMEAN_H
+#define RESPONSEHISTORYMEAN_H
+
+// MOOSE includes
+#include "GeneralVectorPostprocessor.h"
+
+// Forward Declarations
+class ResponseHistoryMean;
+
+template <>
+InputParameters validParams<ResponseHistoryMean>();
+
+/**
+ *  ResponseHistoryMean is a type of NodalVectorPostprocessor computes the mean
+ *  response history given a suite of response histories from the
+ *  ResponseHistoryBuilder, and outputs them into a csv file.
+ */
+class ResponseHistoryMean : public GeneralVectorPostprocessor
+{
+public:
+  ResponseHistoryMean(const InputParameters & parameters);
+  virtual void initialize() override;
+  virtual void initialSetup() override;
+  virtual void execute() override;
+
+protected:
```

Comment:
You have too much stuff here, some of which you don't even use. You need:

const VectorPostprocessorValue & _history_time;
const VectorPostprocessorValue & _history_mean;
const ResponseHistoryBuilder & _builder;

Reply:
Removed the unwanted stuff.